src/Plot.elm

```diff
@@ -0,0 +1,107 @@
+module Plot where
+
+import Svg exposing (svg, Svg)
+import Private.Extras.SvgAttributes exposing (width, height)
+import Private.Dimensions exposing (Dimensions)
+import Private.Margins as Margins exposing (Margins)
+import Private.BoundingBox as BoundingBox exposing (BoundingBox)
+import Private.Scale.Utils as Scale
+import Private.Scale exposing (Scale)
+import Plot.Axis as Axis exposing (Axis)
+import Private.Bars as Bars exposing (Orient)
+import Private.Axis.View as AxisView
+import Private.Title as Title
+
+type alias Plot =
+  { dimensions: Dimensions
+  , margins: Margins
+  , svgs: List (BoundingBox -> List Svg)
+  , eventHandlers: List (BoundingBox -> Svg.Attribute)
+  , attrs: List Svg.Attribute
+  , title : Title.Model
+  }
+
+createPlot : Float -> Float -> Plot
+createPlot w h =
+  { dimensions = { width = w, height = h }
+  , margins = Margins.init
+  , svgs = []
+  , eventHandlers = []
+  , attrs = [width w, height h]
+  , title = Title.init
+  }
+
+addTitle : String -> List Svg.Attribute -> Plot -> Plot
+addTitle title attrs plot =
+  { plot | title = Title.create title attrs }
+
+attributes : List Svg.Attribute -> Plot -> Plot
+attributes attrs plot =
+  { plot | attrs = plot.attrs ++ attrs }
+
+margins : Margins -> Plot -> Plot
+margins m plot =
+  { plot | margins = m }
+
+addVerticalBars : List a -> (a -> b) -> (a -> c) -> Scale x b -> Scale y c -> List Svg.Attribute -> Plot -> Plot
+addVerticalBars points getX getY xScale yScale attrs plot =
+  addBars points getX getY xScale yScale attrs Bars.Vertical plot
+
+addHorizontalBars : List a -> (a -> b) -> (a -> c) -> Scale x b -> Scale y c -> List Svg.Attribute -> Plot -> Plot
+addHorizontalBars points getX getY xScale yScale attrs plot =
+  addBars points getX getY xScale yScale attrs Bars.Horizontal plot
+
+addAxis : Axis a b -> Plot -> Plot
+addAxis axis plot =
+  let
+    svg = \bBox ->
+      let
+        scale =
+          if axis.orient == Axis.Top || axis.orient == Axis.Bottom then
+            Scale.rescaleX bBox axis.scale
+          else
+            Scale.rescaleY bBox axis.scale
+        a = { axis
+              | scale = scale
+              , boundingBox = bBox
+            }
+      in
+        [ AxisView.toSvg a ]
+  in
+    addSvg svg plot
+
+toSvg : Plot -> Svg
+toSvg plot =
```

Comment:
Minor nitpick, but I think this function should live at the end of the file since it is the last thing you call after using all of the other function to configure the `Plot`.